catalog/core/catalog-core-migratable/src/test/java/org/codice/ddf/catalog/migratable/impl/MigrationTaskManagerTest.java

```diff
@@ -0,0 +1,253 @@
+        verify(mockExecutor).shutdownNow();
+    }
+
+    @Test
+    public void testExportFinishedTimeoutShutdown() throws Exception {
+        when(mockExecutor.awaitTermination(1L, TimeUnit.MINUTES)).thenReturn(false);
+
+        taskManager.exportFinish();
+
+        verify(mockExecutor).shutdown();
+        verify(mockExecutor).shutdownNow();
+    }
+
+    @Test(expected = MigrationException.class)
+    public void testExportFinishedErrorOccurred() throws Exception {
+        taskManager.getAtomicReference()
+                .set(mockThrowable);
+        taskManager.exportFinish();
+    }
+
+    @Test
+    public void testExportFinishedNoError() throws Exception {
+        taskManager.getAtomicReference()
+                .set(null);
+        taskManager.exportFinish();
+        //verify(mockAtomicRef).get(); // Can't execute this line - get() is final
+    }
+
+    @Test(expected = MigrationException.class)
+    public void testExportMetacardQueryWithError() throws Exception {
+        List<Result> testResults = new ArrayList<>();
+        for (int i = 0; i < TEST_RESULTS_LIST_SIZE; i++) {
+            testResults.add(new ResultImpl());
+        }
+
+        when(mockExecutor.submit(any(CatalogWriterCallable.class))).thenReturn(mockFutureInstance);
+
+        taskManager.getAtomicReference()
+                .set(mockThrowable);
+        taskManager.exportMetacardQuery(testResults, TEST_GROUP_COUNT);
+    }
+
+    @Test
+    public void testExportMetacardQuerySubmit() throws Exception {
+        List<Result> testResults = new ArrayList<>();
+        for (int i = 0; i < TEST_RESULTS_LIST_SIZE; i++) {
+            testResults.add(new ResultImpl());
+        }
+
+        config.setExportCardsPerFile(TEST_FILE_SIZE);
+
+        List<CatalogWriterCallable> writerCallableList = new ArrayList<>();
+
+        when(mockExecutor.submit(any(CatalogWriterCallable.class))).thenReturn(mockFutureInstance);
+
+        for (int i = 0; i < testResults.size(); i += config.getExportCardsPerFile()) {
+            File expectedFile = config.getExportPath()
+                    .resolve(taskManager.makeFileName(TEST_GROUP_COUNT, i))
+                    .toFile();
+            List<Result> expectedResults = testResults.subList(i,
+                    Math.min((i + config.getExportCardsPerFile()), testResults.size()));
+            CatalogWriterCallable writerCallable = new CatalogWriterCallable(expectedFile,
+                    expectedResults,
+                    mockFileWriter);
+            writerCallableList.add(writerCallable);
+        }
+
+        taskManager.exportMetacardQuery(testResults, TEST_GROUP_COUNT);
+
+        assertTrue(writerCallableList.equals(taskManager.getCallables()));
+
+        int fileCount = TEST_RESULTS_LIST_SIZE / TEST_FILE_SIZE;
+        if (TEST_FUTURES_LIST_SIZE % TEST_FILE_SIZE > 0) {
+            fileCount++;
+        }
+
+        verify(mockExecutor, times(fileCount)).submit(any(CatalogWriterCallable.class));
+    }
```

Comment:
We also need to test the edge cases, e.g., `exportMetacardQuery()` called with an empty result set, with a result set the same size as the file size, when the first result set is processed successfully but the second one fails, etc.

We also need to verify that the Future callback is added, otherwise the whole error handling strategy would fail.